Formula/ship.rb

```diff
@@ -15,6 +15,8 @@ class Ship < Formula
   depends_on "node" => :build
   depends_on "yarn" => :build
 
+  depends_on :macos => :high_sierra # Ship fails to build with Golang 1.12 on Sierra
```

Comment:
Easier but wrong :(

Reply:
I maintain the upstream, can I say it's not supported?

In all seriousness, there seems to be a compiler bug causing failures with Go 1.12 and MacOS Sierra. I could have builds depend on old versions of Golang, but that has other issues, and it feels wrong to degrade performance on the most commonly used platforms in order to help the least commonly used.

Unless there's a way to conditionally add build dependencies based on the OS?

Reply:
@fxcoudert would you prefer for me to just require Go 1.10 until the compiler issue is fixed?

Reply:
I'm going to merge as-is, please try in the next release to remove this requirement :)

Reply:
I'm not sure the compiler issue will be fixed by then (in fact, I'm sure it won't since I just opened a PR for the next release) but I'll remove the requirement as soon as I can!